higher-order-functions/min-max-sorted.md

```diff
@@ -0,0 +1,342 @@
+# Passing Functions as Arguments: Min, Max, and Sort
+
+## Learning Goals
+
+* Demonstrate how to pass a function as an argument
+* Describe how to use the key parameter in the `sorted` function to sort a list of dictionaries by a specific key
+* Describe how to use the key parameter in the `min` function to find the minimum in a list of dictionaries by a specific key
+
+## Introduction
+
+Imagine we have a list of movie titles we'd like to sort alphabetically. Python provides a couple convenient functions for this task. The [`sort`](https://docs.python.org/3/howto/sorting.html#sorting-basics) method can be called on a list and sorts the list in place. The `sorted` function takes a list as a parameter and returns the sorted list.  
+
+But what if we have a list of dictionaries representing movies that we want to sort by title, or perhaps by rating. Or perhaps we want to find the most recent movie or oldest movie in a list.
+
+Python provides the `key` parameter for `min`, `max`, and `sorted` functions, just to name a few. The `key` parameters takes a function that provides the key to use to indicate which attribute to use for finding the minimum, finding the maximum, or to sort. 
+
+## Movie Example
+
+### List of Movie Titles
+
+Let's consider to the following list of movie titles.
+
+```py
+movie_titles = [
+    "Recursion",
+    "Zero Dark Python",
+    "2 JavaScript 2 React",
+    "Javascript 3: VS Code Lint"
+]
+```
+
+To sort the titles alphabetically in place we can use `sort` method:
+
+```py
+movie_titles.sort()
+
+# movie_titles = [
+#   '2 JavaScript 2 React', 
+#   'Javascript 3: VS Code Lint', 
+#   'Recursion',
+#   'Zero Dark Python'
+# ]
+```
+
+Alteratively, we can use the function `sorted` to return the list sorted alphabetically:
```

Comment:
Since we're explicit about the `sort` method sorting a list in-place, we may want to be explicit that `sorted` returns a new array:

```suggestion
Alternatively, we can use the function `sorted` to return a copy of the list sorted alphabetically:
```